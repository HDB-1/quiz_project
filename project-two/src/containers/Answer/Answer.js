import React, { Component } from "react";

export default function Answer (props) {
  
    return (
      <div>
        <input id={props.id} type="button" value={atob(props.answer)} name="answer" onClick={() => props.submit(props.answer)}/>
      </div>
    )
  }
