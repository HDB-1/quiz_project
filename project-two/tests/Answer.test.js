import React from 'react';
import { shallow } from 'enzyme';
import Answer from '../src/containers/Answer/Answer'

describe('Shallow Answer', () => {
  let wrapper;

  beforeEach(() => wrapper = shallow(<Answer answer={'Answer 1'}/>)); // this automatically tests for rendering without crashing.
  it("Should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should render a radio button with the correct information', () => {
    expect(wrapper.find('input').length).toEqual(1);
    expect(wrapper.find('input').find({type: 'radio'}));
    expect(wrapper.find('input').text()).toEqual('Answer 1')
  });
});