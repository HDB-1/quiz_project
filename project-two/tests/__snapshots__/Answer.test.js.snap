// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Answer Should render correctly 1`] = `
<div>
  <input
    name="answer"
    type="radio"
  />
  <label>
    Answer 1
  </label>
</div>
`;
